src/routes/routes.js

```diff
@@ -5,13 +5,15 @@ const Search = require('../search/search')
 const index = require('./index')
 const content = require('./content')
 const author = require('./author')
+const course = require('./course')
 
 const search = new Search()
 
 const allRoutes = {
   index: index(Course),
   content: content(Content, search),
   author: author(Author, search, Content),
+  course: course(Course, search),
```

Comment:
Todas as rotas recebem o mesmo objeto `search`. Não sei exatamente que efeitos colaterais isso pode causar, mas acho que podem acontecer problemas estranhos por causa disso. Acho que enviar o construtor de Search para rota seria uma opção melhor.

Reply:
O comentário ficou confuso, desculpem. A ideia é que, ao invés de criar um único objeto search e enviá-lo para todas as rotas, seria mais interessante enviar ou o construtor de `Search` ou objetos separados para cada rota, como uma forma de evitar efeitos colaterais inesperados. Por exemplo:

Ou enviamos o construtor de Search:
 ```javascript
const allRoutes = {
   index: index(Course),
   content: content(Content, Search),
   author: author(Author, Search, Content),
   course: course(Course, Search)
```

O que significa que cada rota deve instanciar seus objetos `Search`.

Ou enviamos objetos `search`individuais para cada rota, evitando que uma rota altere as informações sendo utilizadas por outra rota, já que todas estão usando o mesmo search, acredito que isso pode acontecer em algum momento.

 ```javascript
const allRoutes = {
   index: index(Course),
   content: content(Content, new Search()),
   author: author(Author, new Search(), Content),
   course: course(Course, new Search())
```

Reply:
A primeira maneira não funciona (aparece Search.setFilter is not a function), mas a segunda dá certo